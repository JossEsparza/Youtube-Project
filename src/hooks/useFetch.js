import { useState, useEffect } from 'react';

export const useFetch = (url) => {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    let controller = new AbortController();

    const fetchData = async () => {
      setLoading(true);

      try {
        const response = await fetch(url, { signal: controller.signal });
        const result = await response.json();
        setData(result);
      } catch (error) {
        setError(error);
      }
      setLoading(false);
    };
    fetchData();

    return () => {
      controller.abort();
    };
  }, [url]);

  return {
    data,
    loading,
    error,
  };
};
