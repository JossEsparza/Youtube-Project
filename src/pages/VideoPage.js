import React from 'react';
// import { useFetch } from '../hooks/useFetch';
// import { useLocation } from 'react-router-dom';
import RelatedVideos from '../components/related_videos/RelatedVideos';
import { GrChannel } from 'react-icons/gr';
import './VideoPage.scss';

const VideoPage = () => {
  const data = require('../mocks/youtube-specific-video-mock.json');
  const error = 0;
  const loading = 0;

  // const location = useLocation();

  // const { data, loading, error } = useFetch(
  //   'https://www.googleapis.com/youtube/v3/videos?part=id, snippet,player&key=' +
  //     process.env.REACT_APP_YOUTUBE_API_KEY +
  //     '&id=' +
  //     location.pathname.substring(1)
  // );

  const { items } = data;

  if (error) {
    return <p>{error.message}</p>;
  }

  return (
    <div className="video_page_container">
      {loading ? (
        <p>Loading!</p>
      ) : (
        <div className="video_page_info_container">
          <div className="video_player">
            <iframe
              title={items[0].id}
              width="100%"
              height="100%"
              src={`//www.youtube.com/embed/${items[0].id}`}
              frameBorder="0"
              allowFullScreen
              allow={
                'accelerometer; autoplay; clipboard-write;' +
                'encrypted-media; gyroscope; picture-in-picture'
              }
            />
            <h3>{items[0].snippet.title}</h3>
            <p className="channel_icon">
              <GrChannel /> {items[0].snippet.channelTitle}
            </p>
            <span>Description: {items[0].snippet.description}</span>
          </div>
          <RelatedVideos />
        </div>
      )}
    </div>
  );
};

export default VideoPage;
