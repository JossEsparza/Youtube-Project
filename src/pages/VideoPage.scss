.video_page_info_container {
  display: flex;
  flex-direction: row;
  margin-top: 20px;
}

.video_player {
  justify-items: center;
  margin-left: 40px;
  margin-right: 40px;
}

p {
  font-weight: bold;
  margin-bottom: -1px;
}

.video_page_container {
  height: 100vh;
}

.dark {
  h4 {
    color: #cbcccd;
  }
  p {
    color: #a3a3a6;
  }

  .video_page_container {
    background-color: #1b1d21;
  }
}
