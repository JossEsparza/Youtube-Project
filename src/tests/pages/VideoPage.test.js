import { mount } from 'enzyme';
import { Route, MemoryRouter } from 'react-router-dom';
import { useFetch } from '../../hooks/useFetch';
import VideoPage from '../../pages/VideoPage';
jest.mock('../../hooks/useFetch');

describe('Tests the Video Page', () => {
  test('should be displayed correctly when loading is true', () => {
    const mockLocation = {
      pathname: '/',
      search: '',
    };
    useFetch.mockReturnValue({
      data: {},
      loading: true,
      error: null,
    });
    const wrapper = mount(
      <MemoryRouter history={mockLocation}>
        <Route path="/:video_id">
          <VideoPage />
        </Route>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should be displayed correctly when loading is false', () => {
    const mockLocation = {
      pathname: '/',
      search: '',
    };
    useFetch.mockReturnValue({
      data: {
        items: [
          {
            snippet: {
              title: 'my first video',
              thumbnails: {
                default: {
                  url: 'https://i.ytimg.com/vi/nmXMgqjQzls/default.jpg',
                },
              },
            },
          },
        ],
      },
      loading: false,
      error: null,
    });
    const wrapper = mount(
      <MemoryRouter history={mockLocation}>
        <Route path="/:video_id">
          <VideoPage />
        </Route>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
