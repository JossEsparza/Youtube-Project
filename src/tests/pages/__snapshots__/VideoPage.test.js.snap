// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Video Page should be displayed correctly when loading is false 1`] = `ReactWrapper {}`;

exports[`Tests the Video Page should be displayed correctly when loading is true 1`] = `ReactWrapper {}`;
